### PR TITLE
PICARD-1288: Fix artist genre fallback

### PR DESCRIPTION
**Pull request by phw**
# Summary
Fix artist genre fallback if "Fall back on album's artists genres" is active

* This is a…
    * [x] Bug fix
    * [ ] Feature addition
    * [ ] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem
The artists tags where not read and hence could not be used as a fallback if no tags are set on recording, release or release-group level

* JIRA ticket (_optional_): [PICARD-1288](https://tickets.metabrainz.org/browse/PICARD-1288)

# Solution
Together with [MBS-9400](https://tickets.metabrainz.org/browse/MBS-9400) this fixes the issue.

# Action

### CODE REVIEW

File: picard/mbjson.py [hunk 1]
```diff
@@ -366,7 +366,8 @@ def recording_to_metadata(node, m, track=None):
             # set tags from artists
             if track:
                 for artist in value:
-                    track.append_track_artist(artist['artist']['id'])
+                    artist_obj = track.append_track_artist(artist['artist']['id'])
+                    add_genres_from_node(artist['artist'], artist_obj)
```
Comment (zas) [inline picard/mbjson.py]: I think it'd be more readable to use another intermediate variable, something like:

```python
                for artist in value:
                    art = artist['artist']
                    artist_obj = track.append_track_artist(art['id'])
                    add_genres_from_node(art, artist_obj)
```

File: picard/mbjson.py [hunk 2]
```diff
@@ -448,7 +449,8 @@ def release_to_metadata(node, m, album=None):
             # set tags from artists
             if album is not None:
                 for artist in value:
-                    album.append_album_artist(artist['artist']['id'])
```
Comment (zas) [inline picard/mbjson.py]: Same here

**Comment by zas** [comment]
Tests are failing though

File: picard/mbjson.py (hunk 1 above)
Comment (phw) [reply picard/mbjson.py]: Done, also added unit test

**Comment by phw** [comment]
> Tests are failing though

Yes, wrong indentation of one line. Also fixed

**Pull request merged**